lib/groups.js

```diff
@@ -104,28 +139,33 @@ module.exports.joinGroup = function (joinCode, assessmentId, userId, authnUserId
 };
 
 module.exports.createGroup = function (groupName, assessmentId, userId, authnUserId, callback) {
-  const params = {
+  let params = {
     assessment_id: assessmentId,
     user_id: userId,
     authn_user_id: authnUserId,
     group_name: groupName,
+    group_role_id: null,
   };
   //alphanumeric characters only
   let invalidGroupName = true;
   const letters = /^[0-9a-zA-Z]+$/;
   if (groupName.length <= 30 && groupName.match(letters)) {
     invalidGroupName = false;
-    sqldb.query(sql.create_group, params, function (err, _result) {
-      if (!err) {
-        callback(null, true);
-      } else {
-        sqldb.query(sql.get_config_info, params, function (err, result) {
-          if (ERR(err, callback)) return;
-          const permissions = result.rows[0];
-          const uniqueGroupName = true;
-          callback(null, false, uniqueGroupName, null, permissions);
-        });
-      }
+    sqldb.query(sql.get_creator_role, params, function (err, result) {
+      if (ERR(err, callback)) return;
+      params.group_role_id = result.rows[0].id;
+      sqldb.query(sql.create_group, params, function (err, _result) {
+        if (!err) {
```

Comment:
I know this is existing code that you didn't touch, but I don't understand what's happening here: is `create_group` expected to error? Why do we `get_config_info` if there was an error?

Reply:
`create_group` will error if the group name is already taken. If the group name is taken or invalid, the page needs to re-render with the appropriate errors, so the callback function ends up calling `res.render(...)`. `get_config_info` is called to get `permissions` which is needed in `res.locals` for the page to load.

This is a mini version of the problem that Renzo mentioned [here](https://github.com/PrairieLearn/PrairieLearn/pull/6109#discussion_r1037784714), where we can't use a POST/redirect/GET pattern because the GET call can't remember the errors to display.